### PR TITLE
paradox javadoc and scaladoc directives for cluster-client.md

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #22904

paradox pages generated in my firebase demo site:
https://richard-akka-doc.firebaseapp.com/cluster-client.html

Confirmed hyperlinks work fine.

Although lines are getting very long horizontally, I am not adding line breaks as that would make reviewing the diff very difficult, especially if I am adding new commits from here.
Once everything looks ok and we still want line breaks, I can add them in the end.

### CODE REVIEW

File: akka-docs/src/main/paradox/cluster-client.md [hunk 1]
```diff
@@ -23,23 +23,23 @@ to `remote` or `cluster` when using
...
+You can send messages via the @scala[@scaladoc[`ClusterClient`](akka.cluster.client.ClusterClient)]@java[@javadoc[`ClusterClient`](akka.cluster.client.ClusterClient)] to any actor in the cluster that is registered
+in the @scala[@scaladoc[`DistributedPubSubMediator`](akka.cluster.pubsub.DistributedPubSubMediator)]@java[@javadoc[`DistributedPubSubMediator`](akka.cluster.pubsub.DistributedPubSubMediator)] used by the @scala[@scaladoc[`ClusterReceptionist`](akka.cluster.client.ClusterReceptionist)]@java[@javadoc[`ClusterReceptionist`](akka.cluster.client.ClusterReceptionist)].
+The @scala[@scaladoc[`ClusterClientReceptionist`](akka.cluster.client.ClusterClientReceptionist)]@java[@javadoc[`ClusterClientReceptionist`](akka.cluster.client.ClusterClientReceptionist)] provides methods for registration of actors that
 should be reachable from the client. Messages are wrapped in `ClusterClient.Send`,
-`ClusterClient.SendToAll` or `ClusterClient.Publish`.
+@scala[@scaladoc[`ClusterClient.SendToAll`](akka.cluster.client.ClusterClient$)]@java[`ClusterClient.SendToAll`] or @scala[@scaladoc[`ClusterClient.Publish`](akka.cluster.client.ClusterClient$)]@java[`ClusterClient.Publish`].
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/cluster-client.md]: ### scala
scaladoc is linked to:
https://doc.akka.io/api/akka/2.5/akka/cluster/client/ClusterClient$

As the URL for `SendToAll` includes the return type, and messy (also the part after `#`might be subject to change in future?).
https://doc.akka.io/api/akka/2.5/akka/cluster/client/ClusterClient$.html#SendextendsProductwithSerializable

### java
javadoc link is not added as there is no way to get it work. 
The correct URL is:
https://doc.akka.io/japi/akka/2.5/akka/cluster/client/ClusterClient.SendToAll.html,

but `@javadoc[...](akka.cluster.client.ClusterClient.Publish)`  will be linked to 
.....client/ClusterClient/SendToAll.html (`/` after ClusterClient) 
which doesn't exist.

File: akka-docs/src/main/paradox/cluster-client.md [hunk 2]
```diff
@@ -63,18 +63,18 @@ what clients are connected.
 Response messages from the destination actor are tunneled via the receptionist
 to avoid inbound connections from other cluster nodes to the client:
 
-* @scala[`sender()`,] @java[`getSender()`,] as seen by the destination actor, is not the client itself, 
+* @scala[@scaladoc[`sender()`,](akka.actor.Actor)]@java[@javadoc[`getSender()`,](akka.actor.Actor)], as seen by the destination actor, is not the client itself,
   but the receptionist
-* @scala[`sender()`] @java[`getSender()`] of the response messages, sent back from the destination and seen by the client, 
+* @scala[@scaladoc[`sender()`](akka.actor.Actor)] @java[@javadoc[`getSender()`](akka.actor.Actor)] of the response messages, sent back from the destination and seen by the client,
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/cluster-client.md]: Similar to [this comment](https://github.com/akka/akka/pull/24524/files#r167429853), the URL for the `sender()` Scala method includes the return type:
https://doc.akka.io/api/akka/2.5/akka/actor/Actor.html#self:akka.actor.ActorRef

So I avoid this messy URL and link it to the `Actor` trait Scaladoc.

File: akka-docs/src/main/paradox/cluster-client.md [hunk 3]
```diff
@@ -63,18 +63,18 @@ what clients are connected.
 Response messages from the destination actor are tunneled via the receptionist
 to avoid inbound connections from other cluster nodes to the client:
 
-* @scala[`sender()`,] @java[`getSender()`,] as seen by the destination actor, is not the client itself, 
+* @scala[@scaladoc[`sender()`,](akka.actor.Actor)]@java[@javadoc[`getSender()`,](akka.actor.Actor)] as seen by the destination actor, is not the client itself,
   but the receptionist
-* @scala[`sender()`] @java[`getSender()`] of the response messages, sent back from the destination and seen by the client, 
+* @scala[@scaladoc[`sender()`](akka.actor.Actor)] @java[@javadoc[`getSender()`](akka.actor.Actor)] of the response messages, sent back from the destination and seen by the client,
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/cluster-client.md]: Similar to [this comment](https://github.com/akka/akka/pull/24524/files#r167429853), the URL for the `sender()` Scala method includes the return type:
https://doc.akka.io/api/akka/2.5/akka/actor/Actor.html#self:akka.actor.ActorRef

So I avoid this messy URL and just link it to the `Actor` trait Scaladoc.

File: akka-docs/src/main/paradox/cluster-client.md [hunk 4]
```diff
@@ -63,18 +63,18 @@ what clients are connected.
 Response messages from the destination actor are tunneled via the receptionist
 to avoid inbound connections from other cluster nodes to the client:
 
-* @scala[`sender()`,] @java[`getSender()`,] as seen by the destination actor, is not the client itself, 
+* @scala[@scaladoc[`sender()`](akka.actor.Actor)]@java[@javadoc[`getSender()`](akka.actor.Actor)], as seen by the destination actor, is not the client itself,
   but the receptionist
-* @scala[`sender()`] @java[`getSender()`] of the response messages, sent back from the destination and seen by the client, 
+* @scala[@scaladoc[`sender()`](akka.actor.Actor)] @java[@javadoc[`getSender()`](akka.actor.Actor)] of the response messages, sent back from the destination and seen by the client,
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/cluster-client.md]: Similar to [this comment](https://github.com/akka/akka/pull/24524/files#r167429853), the URL for the `sender()` Scala method includes the return type:
https://doc.akka.io/api/akka/2.5/akka/actor/Actor.html#self:akka.actor.ActorRef

So I avoid this messy URL and just link it to the `Actor` trait Scaladoc.

**Comment by richardimaoka** [comment]
ready for review

File: akka-docs/src/main/paradox/cluster-client.md [hunk 5]
```diff
@@ -23,23 +23,23 @@ to `remote` or `cluster` when using
...
 
 The receptionist is supposed to be started on all nodes, or all nodes with specified role,
-in the cluster. The receptionist can be started with the `ClusterClientReceptionist` extension
+in the cluster. The receptionist can be started with the @scala[@scaladoc[`ClusterClientReceptionist`](akka.cluster.client.ClusterClientReceptionist)]@java[@javadoc[`ClusterClientReceptionist`](akka.cluster.client.ClusterClientReceptionist)] extension
 or as an ordinary actor.
 
-You can send messages via the `ClusterClient` to any actor in the cluster that is registered
-in the `DistributedPubSubMediator` used by the `ClusterReceptionist`.
-The `ClusterClientReceptionist` provides methods for registration of actors that
+You can send messages via the @scala[@scaladoc[`ClusterClient`](akka.cluster.client.ClusterClient)]@java[@javadoc[`ClusterClient`](akka.cluster.client.ClusterClient)] to any actor in the cluster that is registered
```
Comment (ktoso) [inline akka-docs/src/main/paradox/cluster-client.md]: Would be so nice to have a magical directive for that, perhaps understanding scaladsl/javadsl hmm

**Review by ktoso — APPROVED** [review]
Seems correct, thanks!

File: akka-docs/src/main/paradox/cluster-client.md (hunk 5 above)
Comment (raboof) [reply akka-docs/src/main/paradox/cluster-client.md]: #24426 :)

**Pull request merged**